### PR TITLE
Add Triangulation::all_reference_cells_are_simplex() and ::is_mixed_mesh()

### PR DESCRIPTION
**Pull request by peterrum**
as suggested in https://github.com/dealii/dealii/pull/13042#discussion_r764201436

### CODE REVIEW

File: source/grid/tria.cc [hunk 1]
```diff
@@ -14655,7 +14655,36 @@ Triangulation<dim, spacedim>::all_reference_cells_are_hyper_cube() const
...
+bool
+Triangulation<dim, spacedim>::is_mixed_mesh() const
+{
+  Assert(this->reference_cells.size() > 0,
+         ExcMessage("You can't ask about the kinds of reference "
+                    "cells used by this triangulation if the "
+                    "triangulation doesn't yet have any cells in it."));
+  return reference_cells.size() > 1 ||
+         ((reference_cells[0].is_hyper_cube() ||
+           reference_cells[0].is_simplex()) == false);
```
Comment (bangerth) [inline source/grid/tria.cc:14687]: This is unnecessarily hard to read. Write it as
```suggestion
         ((reference_cells[0].is_hyper_cube() == false) &&
           (reference_cells[0].is_simplex() == false));
```

**Review by bangerth — COMMENTED** [review]
OK if you fix the one typo and the one condition.

**Comment by peterrum** [comment]
@bangerth Thanks for the review. I have applied the changes!

**Review by bangerth — APPROVED** [review]

**Review by kronbichler — APPROVED** [review]

**Comment by nfehn** [comment]
I just wanted to ask whether the spelling is correct, i.e. should it be `_are_simplicial()` or `are_simplices()`?

**Comment by peterrum** [comment]
@nfehn  I have selected the name this way, since there is already a function called `all_reference_cells_are_hyper_cube()` (this was already part of the last release). I don't mind to change the names but let's first wait for opinions by others!

**Comment by bangerth** [comment]
I think all of these are acceptable as far as language is concerned. The way most people would read the name is "[are] all reference cells simplex [cells]?" In that case, "simplex cells" is a compound noun. In "[are] all reference cells simplicial [cells]?" the word "simplicial" is an adjective.

If we had to reinvent names, `all_reference_cells_are_hyper_cubes/simplices()"` might have been a better name, but it's hard to go back in time.

**Pull request merged**